draft/0006-pat.md

```diff
@@ -0,0 +1,125 @@
+---
+Start date: 2024-08-15
+Author(s): wangsijie
+Revision: 1
+---
+
+# Personal Access Token (PAT)
+
+## Overview
+
+This RFC proposes an extension to Logto's authentication system to support Personal Access Tokens (PATs). This feature aims to provide users with a secure and convenient way to authenticate automated processes and integrations. PATs are long-lived tokens that can be used by non-interactive clients to exchange for short-lived access tokens through token exchange endpoints first introduced in RFC-0005 impersonation.
+
+## Motivation
+
+The current authentication flow in Logto is optimized for interactive user sessions. However, there are scenarios where users need to authenticate non-interactive processes or third-party integrations. These include:
+
+1. Automated scripts or bots that perform actions on behalf of the user.
+2. Integration with external tools and services that require API access.
+3. Long-running processes that outlast typical session durations.
+
+## Personal Access Token (PAT) Creation
+
+The client makes a post request to the tenant's Management API `/api/users/:userId/personal-access-tokens` to create a PAT for the user. The following parameters are included in the HTTP request entity-body using the `application/json` format.
```

Comment:
Should we also have RUD operations for PATs? Seems PAT is a kind of long-lived secret/credential.

Reply:
Yes we do have RUD operations, but I think there is no need to include them in this RFC, keep simple, focus to the core flow.